modules/shared/interpreter/src/main/scala/almond/interpreter/messagehandlers/InterpreterMessageHandlers.scala

```diff
@@ -168,6 +170,28 @@ final case class InterpreterMessageHandlers(
       inspectHandler
     )
 
+  def shutdownHandler: MessageHandler = {
+    // v5.3 spec states "The request can be sent on either the control or shell channels.".
+    def forChannel(c: Channel): MessageHandler = {
+      MessageHandler(c, Shutdown.requestType) { message =>
+        val reply = IO {
+          val reply = message.reply(Shutdown.replyType, Shutdown.Reply(message.content.restart))
```

Comment:
The `IO { … }` is unnecessary here, it wraps a "pure" value (what is passed to it has no side effect, and always evaluates to the same value). Dropping it, one can just create a stream with `Stream(reply)`.

And `Message[_]` has an `onStream` method that can help here, like
```scala
message
  .reply(Shutdown.replyType, Shutdown.Reply(message.content.restart))
  .streamOn(c)
```
provides directly what `Stream.eval(reply)` was giving.